packages/astro/src/core/errors/errors-data.ts

```diff
@@ -1272,5 +1272,21 @@ export const UnsupportedConfigTransformError = {
 	hint: 'See the devalue library for all supported types: https://github.com/rich-harris/devalue',
 } satisfies ErrorData;
 
+export const MissingLocale = {
+	name: 'MissingLocaleError',
+	title: 'The provided locale does not exist.',
+	message: (locale: string, locales: string[]) => {
+		return `The locale \`${locale}\` does not exist in the configured locales. Available locales: ${locales.join(
+			', '
+		)}.`;
+	},
+} satisfies ErrorData;
+
+export const CantRenderPage = {
+	name: 'CantRenderPage',
+	title: "Astro can't render the route.",
+	message: "You tried to render a route that isn't a redirect and doesn't have any component.",
```

Comment:
Hmm... The hint could maybe be something like:

> If you expect to find a route here, this may be an Astro bug. Please file an issue/restart the dev server/...

Or something like that. It would be helpful to stop a user from checking and double checking all their code if there's nothing that they can actually do here. Because I think the natural thing to do when you get an error is try to change something in your code, so if we can stop them from doing that, that's helpful.

Reply:
As for the message itself, maybe:

> You attempted to render a route that does not exist. There is no file or redirect associated with this route."